syntax = "proto3";

package wpi.proto;

option java_package = "edu.wpi.first.math.proto";

message ProtobufTranslation3d {
  double x_meters = 1;
  double y_meters = 2;
  double z_meters = 3;
}

message ProtobufQuaternion {
  double w = 1;
  double x = 2;
  double y = 3;
  double z = 4;
}

message ProtobufRotation3d {
  ProtobufQuaternion q = 1;
}

message ProtobufPose3d {
  ProtobufTranslation3d translation = 1;
  ProtobufRotation3d rotation = 2;
}

message ProtobufTransform3d {
  ProtobufTranslation3d translation = 1;
  ProtobufRotation3d rotation = 2;
}

message ProtobufTwist3d {
  double dx_meters = 1;
  double dy_meters = 2;
  double dz_meters = 3;
  double rx_radians = 4;
  double ry_radians = 5;
  double rz_radians = 6;
}
